src/app/common/user.service.spec.js

```diff
@@ -0,0 +1,159 @@
+(function() {
+  'use strict';
+
+  describe('userService ', function(){
+    var self, controller, scope, userService, user;
+
+    beforeEach( module('healthGuide'));
+    beforeEach( inject( function (_userService_) {
+      userService = _userService_;
+      user = jQuery.extend(true, {}, userService.user);
+    }));
+    afterEach(function(){
+      userService.user = user;
+    });
+
+
+    describe('user', function(){
+      it('is Object', function() {
+        expect(userService.user).toEqual(jasmine.any(Object));
+      });
+      it('has need property', function(){
+        expect(Object.keys(userService.user)).toContain('name' && 'nutrition'&& 'activity' && 'sleep');
+      });
+
+      it('most init values are falsy', function() {
+        expect(userService.user.BMI).toBeNull();
+        expect(userService.user.BMR).toBeNull();
+        expect(userService.user.dailyCalories).toBeNull();
+        expect(userService.user.sleepDuration).toBeNull();
+
+        _.values(userService.user.sleep).forEach(function (value) {
+          expect(value).toBeFalsy();
+        });
+        _.values(userService.user.activity).forEach(function (value) {
+          expect(value).toBeFalsy();
+        });
+      });
+      it('metrics init values are true', function() {
+        expect(userService.user.nutrition.isWeightKg).toBeTruthy();
+        expect(userService.user.nutrition.isHeightCm).toBeTruthy();
+      });
+    });
+
+    describe('calculate BMI', function() {
+      it('is function', function() {
+        expect(userService.calculateBMI).toEqual(jasmine.any(Function));
+      });
+      it('check formula', function() {
+        userService.user.nutrition.weight = 40;
+        userService.user.nutrition.height = 150;
+        expect(userService.calculateBMI()).toEqual(17.8);
+      });
+      it('Undefined without any input', function() {
+        expect(userService.calculateBMI()).toBeUndefined();
+        userService.user.nutrition.height = 150;
+        expect(userService.calculateBMI()).toBeUndefined();
+        userService.user.nutrition.weight = 40;
+        expect(userService.calculateBMI()).toBeDefined();
```

Comment:
4 different tests:
- user has height, does not have weight
- user has weight, does not have height
- user does not have weight and height
- user has weight and height and formula is calculated properly